src/comments.js

```diff
@@ -462,6 +467,18 @@ function handleMemberExpressionComments(enclosingNode, followingNode, comment) {
   return false;
 }
 
+function handleStaticExpressionComments(enclosingNode, followingNode, comment) {
+  if (
+    enclosingNode &&
+    enclosingNode.static === true
```

Comment:
Outside of small things about your PR, I unfortunately don't think that this is a proper solution. We don't want to force all the comments to be leading if it is static. We only want that if the comment is between `static` and `function` keywords. I'm not really sure how we would do that with the current infrastructure to be honest.